rust/lance-datafusion/src/planner.rs

```diff
@@ -665,6 +685,67 @@ impl Planner {
                 expr: Box::new(self.parse_sql_expr(expr)?),
                 data_type: self.parse_type(data_type)?,
             })),
+            SQLExpr::MapAccess { column, keys } => {
+                let mut expr = self.parse_sql_expr(column)?;
+
+                for key in keys {
+                    let field_access = match &key.key {
+                        SQLExpr::Value(
+                            Value::SingleQuotedString(s) | Value::DoubleQuotedString(s),
+                        ) => GetFieldAccess::NamedStructField {
+                            name: ScalarValue::from(s.as_str()),
+                        },
+                        SQLExpr::JsonAccess { .. } => {
+                            return Err(Error::invalid_input(
+                                "JSON access is not supported",
+                                location!(),
+                            ));
+                        }
+                        key => {
+                            let key = Box::new(self.parse_sql_expr(key)?);
+                            GetFieldAccess::ListIndex { key }
```

Comment:
Do we want to verify this is an integer at some point?

Reply:
I think in theory it could be an expression that resolves to an integer. I'm kind of trusting that it will be verified somewhere down the chain.